### DISCUSSION TITLE
i want to use one rbac series,and now a jenkins access to keycloak as a client,and argocd as another client in keyclloak in the same realm.but i found out i cant use

### DISCUSSION THREAD

**Discussion by mingmingshiliyu** [keycloak/keycloak#25805] [Q&A]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

authorization-services

### Describe the bug

how to integrate jenkins and argocd rbac function together in keycloak, for example,i create  a username/password called jack/123 whether  in keycloak or jenkins,and this username is workable in both jenkins and argocd,how can i do that?  oidc or native client?

### Version

latest

### Expected behavior

create a user in jenkins or keycloak,then the user is workable in argocd ,keycloak,jenkins.

### Actual behavior

i created two clients in keycloak,that`s not suitable

### How to Reproduce?

none

**Comment by souravs17031999** [comment]
@mingmingshiliyu why are you not able to do so ? I don't see enough information here.

Anyway, I will try to explain my previous experience in my projects, I was able to integrate multiple applications to use same keycloak client in a realm. 

What you will have to do is add redirect_uris for all those multiple applications which are trying to integrate and use browser grants for which this client is created.

**Reply by mingmingshiliyu** [reply]
in keycloak,there are many kinds of clients,one is jenkins(clien id),the other is argocd,and also oidc,if I want to integrate argocd and jenkins,create one and only oidc client ? or any practical docs for recommendation?

**Reply by souravs17031999** [reply]
In keycloak, there are only two kinds of clients based on the protocol which you want to use - OIDC and SAML based
https://www.keycloak.org/docs/latest/server_admin/#assembly-managing-clients_server_administration_guide

**Reply by mingmingshiliyu** [reply]
https://www.cnblogs.com/Chinori/p/17091126.html
okay,this doc confused me,it asked me to fill client id as jenkins.i thought the client id was the kind of client.   so I just need to create one client,and multiple systems can use the only client in keycloak,is that true?

**Reply by souravs17031999** [reply]
Indeed
